Kernel/FileSystem/TmpFS.h

```diff
@@ -0,0 +1,74 @@
+#pragma once
+
+#include <Kernel/FileSystem/FileSystem.h>
+#include <Kernel/FileSystem/Inode.h>
+
+class TmpFSInode;
+
+class TmpFS final : public FS {
+    friend class TmpFSInode;
+
+public:
+    virtual ~TmpFS() override;
+    static NonnullRefPtr<TmpFS> create();
+    virtual bool initialize() override;
+
+    virtual const char* class_name() const override { return "TmpFS"; }
+
+    virtual InodeIdentifier root_inode() const override;
+    virtual RefPtr<Inode> get_inode(InodeIdentifier) const override;
+
+    virtual RefPtr<Inode> create_inode(InodeIdentifier parent_id, const String& name, mode_t, off_t size, dev_t, int& error) override;
+    virtual RefPtr<Inode> create_directory(InodeIdentifier parent_id, const String& name, mode_t, int& error) override;
+
+private:
+    TmpFS();
+
+    RefPtr<TmpFSInode> m_root_inode { nullptr };
```

Comment:
No need to initialize this explicitly; a `RefPtr` is always `nullptr` by default.